apis/v1beta1/validation/gateway.go

```diff
@@ -133,3 +135,24 @@ func ValidateListenerNames(listeners []gatewayv1b1.Listener, path *field.Path) f
 	}
 	return errs
 }
+
+// validateHostnameProtocolPort validates that the combination of port, protocol, and name are
+// unique for each listener.
+func validateHostnameProtocolPort(listeners []gatewayv1b1.Listener, path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	hostnameProtocolPortSets := sets.Set[string]{}
+	for i, listener := range listeners {
+		hostname := new(gatewayv1b1.Hostname)
+		if listener.Hostname != nil {
+			hostname = listener.Hostname
+		}
+		protocol := listener.Protocol
+		port := listener.Port
+		hostnameProtocolPort := fmt.Sprintf("%s:%s:%d", *hostname, protocol, port)
+		if hostnameProtocolPortSets.Has(hostnameProtocolPort) {
+			errs = append(errs, field.Forbidden(path.Index(i), fmt.Sprintln("combination of port, protocol, and name must be unique for each listener")))
+		}
+		hostnameProtocolPortSets.Insert(hostnameProtocolPort)
```

Comment:
small nit: consider not calling Insert if the set has the combination already.

Reply:
@LiorLieberman Thanks for your comments, updated.